/* Copyright 2020 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package fedlearner.common;
import "fedlearner/common/common.proto";

message DataBlockRequest {
  uint32 worker_rank = 1;
  //If TM role is follower, trainer request for datablock by block_id from leader.
  string block_id = 2;
};

message DataBlockInfo {
  string block_id = 1;
  string data_path = 2;
  string meta_path = 3; 
}

message DataBlockResponse {
  Status status = 1;
  //返回状态码
  DataBlockInfo data_block_info = 2;
}

service TrainerMasterService {
  rpc RequestDataBlock (DataBlockRequest) returns (DataBlockResponse) {}
}